daemon/list.go

```diff
@@ -382,6 +346,27 @@ func (daemon *Daemon) foldFilter(config *types.ContainerListOptions) (*listConte
 		names:                daemon.nameIndex.GetAll(),
 	}, nil
 }
+func portOp(key string, filter map[nat.Port]bool) func(value string) error {
+	return func(value string) error {
+		if strings.Contains(value, ":") {
+			return fmt.Errorf("filter for '%s' should not contain ':': %v", key, value)
```

Comment:
Didn't change the code inside (just extracted it), but you are right, we could use `%s` for both, I'll update :wink: